### PR TITLE
Added ability to list and download available artifacts

### PR DESCRIPTION
**Pull request by mbergmann**
The current client.get_artifact implementation expects that only one artifact exists for a job.
This change allows the consumer to list all artifacts then download_all or download a specific artifact

### CODE REVIEW

**Comment by mbergmann** [comment]
Ping. Can we get this merged in?

**Comment by arangamani** [comment]
@mbergmann Sorry about the delay. I'll get to it shortly.

File: lib/jenkins_api_client/client.rb [hunk 1]
```diff
@@ -320,6 +299,30 @@ def make_http_request(request, follow_redirect = @follow_redirects)
     end
     protected :make_http_request
 
+    # Downloads a url to a filenam
```
Comment (arangamani) [inline lib/jenkins_api_client/client.rb]: `filenam` -> `filename`

File: lib/jenkins_api_client/client.rb [hunk 2]
```diff
@@ -251,27 +251,6 @@ def inspect
...
-      request = Net::HTTP::Get.new(uri.request_uri)
-      request.basic_auth(@username, @password)
-      response = http.request(request)
-      if response.code == "200"
-        File.write(File.expand_path(filename), response.body)
-      else
-        raise "Couldn't get the artifact"
-      end
-    end
-
```
Comment (arangamani) [inline lib/jenkins_api_client/client.rb]: Can we keep this method so we don't break anybody relying on it?

File: lib/jenkins_api_client/client.rb [hunk 3]
```diff
@@ -320,6 +299,30 @@ def make_http_request(request, follow_redirect = @follow_redirects)
...
+        http.read_timeout = 600
+        http.request request do |response|
+          open filename, 'w' do |io|
+            response.read_body do |chunk|
+              io.write chunk
+            end
+          end
+        end
+      end
+      @logger.info "Stored download as " + filename + "."
```
Comment (arangamani) [inline lib/jenkins_api_client/client.rb]: Let's change t his to `@logger.info "Stored download as #{filename}."` for consistency.

**Comment by katelovescode** [comment]
#251 handles almost all of this functionality, closing as duplicate-ish.

**Pull request closed without merging**